// Copyright 2024 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


message IamAlphaServiceAccount {
  string name = 1;
  string project = 2;
  string unique_id = 3;
  string email = 4;
  string display_name = 5;
  string description = 6;
  string oauth2_client_id = 7;
  IamAlphaServiceAccountActasResources actas_resources = 8;
  bool disabled = 9;
}

message IamAlphaServiceAccountActasResources {
  repeated IamAlphaServiceAccountActasResourcesResources resources = 1;
}

message IamAlphaServiceAccountActasResourcesResources {
  string full_resource_name = 1;
}

message ApplyIamAlphaServiceAccountRequest {
  IamAlphaServiceAccount resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteIamAlphaServiceAccountRequest {
  string service_account_file = 1;
  IamAlphaServiceAccount resource = 2;
}

message ListIamAlphaServiceAccountRequest {
  string service_account_file = 1;
  string Project = 2;
}

message ListIamAlphaServiceAccountResponse {
  repeated IamAlphaServiceAccount items = 1;
}

service IamAlphaServiceAccountService {
  rpc ApplyIamAlphaServiceAccount(ApplyIamAlphaServiceAccountRequest) returns (IamAlphaServiceAccount);
  rpc DeleteIamAlphaServiceAccount(DeleteIamAlphaServiceAccountRequest) returns (google.protobuf.Empty);
  rpc ListIamAlphaServiceAccount(ListIamAlphaServiceAccountRequest) returns (ListIamAlphaServiceAccountResponse);
}
